flytekit/interaction/click_types.py

```diff
@@ -383,6 +383,9 @@ def convert(
             if self._python_type is datetime.date:
                 # Click produces datetime, so converting to date to avoid type mismatch error
                 value = value.date()
+            # If the input matches the default value in the launch plan, serialization can be skipped.
+            if value == param.default:
```

Comment:
should we check if `param` is set to `None` before accessing the default since `param`'s type is set to `Optional`?

Reply:
updated